src/marshmallow/fields.py

```diff
@@ -710,12 +710,15 @@ def _bind_to_schema(self, field_name, schema):
 
     def _serialize(
         self, value, attr, obj, **kwargs
-    ) -> typing.Optional[typing.List[typing.Any]]:
+    ) -> typing.Optional[typing.Iterable[typing.Any]]:
         if value is None:
             return None
-        return [self.inner._serialize(each, attr, obj, **kwargs) for each in value]
+        result = [self.inner._serialize(each, attr, obj, **kwargs) for each in value]
+        if self.serialization_type is list:
+            return result
+        return self.serialization_type(result)  # type: ignore
```

Comment:
I _think_ this is a `mypy` issue. Here's the error I get without the ignore:
```
src/marshmallow/fields.py:719: error: Too many arguments for "Iterable"
```

Reply:
Strange. I'd be happy to get advice from someone more familiar with typing/mypy.